src/components/Menu/MenuItem.tsx

```diff
@@ -113,7 +119,17 @@ export const MenuItem = <C extends MenuItemElement = "button">({
       onClick={onClick}
       disabled={disabled}
     >
-      <Icon width={24} height={24} className={styles.icon} aria-hidden={true} />
+      {iconIsReactElement ? (
+        <div className={styles.icon}>{componentIcon}</div>
```

Comment:
Why not use Radix SlotClone instead of bloating the DOM?

Reply:
Because I never head of Radix SlotClone :)

Reply:
Yes, its the basis for how all of the Radix stuff Compound builds upon works

Reply:
This utility is not currently installed in Compound. Are you talking about the `asChild` pattern ?

Reply:
I tried with `Slot`:
- It works if the children is directly JSX like: `<Slot classname="myClass"><span>content</span></Slot>`
- It doesn't work if I'm doing `<Slot classname="myClass">{myComponent}</Slot>`

Reply:
It empirically works in Tooltip

Reply:
I made it works, I forgot to spread the props of the test component 🙃